### PR TITLE
Refactored quarantine mode tests (closes #358)

### LINKED ISSUE #358: Refactor quarantine mode tests
Quarantine mode auxiliary server must be merged into main functional test server.

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

**Comment by AndreyBelym** [comment]
@testcafe-build-bot \retest

File: test/functional/quarantine-mode-tracker.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+var QuarantineModeTracker = {
```
Comment (inikulin) [inline test/functional/quarantine-mode-tracker.js]: Why not `module.exports = { ...` ?

File: test/functional/fixtures/quarantine-mode/index-test.js [hunk 2]
```diff
@@ -1,96 +1,51 @@
-var expect         = require('chai').expect;
-var http           = require('http');
-var promisifyEvent = require('promisify-event');
-var config         = require('../../config');
+var expect                = require('chai').expect;
+var config                = require('../../config');
+var QuarantineModeTracker = require('../../quarantine-mode-tracker');
```
Comment (inikulin) [inline test/functional/fixtures/quarantine-mode/index-test.js]: It's an object not a ctor. Should be lower camel-case

**Comment by inikulin** [comment]
\r-

**Comment by AndreyBelym** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
lgtm

**Comment by AndreyBelym** [comment]
@VasilyStrelyaev, please, check comments

File: test/functional/fixtures/quarantine-mode/index-test.js [hunk 3]
```diff
@@ -1,96 +1,51 @@
...
+    expect(browserAgents.length).to.equal(config.browsers.length);
 
-    function closeServer () {
-        testServer.close();
-        return promisifyEvent(testServer, 'close');
-    }
-
-    before(function () {
-        return startServer();
+    // NOTE: Last step is 4th:
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/quarantine-mode/index-test.js]: > Last step is 4th

What do you mean? '4 steps in total'?

**Comment by AlexanderMoskovkin** [comment]
lgtm

File: test/functional/fixtures/quarantine-mode/index-test.js (hunk 3 above)
Comment (AndreyBelym) [inline test/functional/fixtures/quarantine-mode/index-test.js]: Yes, there must be 4 steps in total

Comment (VasilyStrelyaev) [inline test/functional/fixtures/quarantine-mode/index-test.js]: hmmm... you are talking about how many times a test is launched in a quarantine, don't you?
so maybe they are not *steps* but rather *attempts* or *runs*?

Comment (AndreyBelym) [inline test/functional/fixtures/quarantine-mode/index-test.js]: *Runs* suits better, indeed

File: test/functional/fixtures/quarantine-mode/index-test.js [hunk 4]
```diff
@@ -1,96 +1,51 @@
...
+    // failing sequence: 1st time - fail, 2nd - pass, 3rd and 4th - fail
+    // passing sequence: 1st time - fail, 2nd, 3rd and 4th - pass
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/quarantine-mode/index-test.js]:
```
    // NOTE: 4 runs in total:
    // failing sequence: 1st run - fail, 2nd - pass, 3rd and 4th - fail
    // passing sequence: 1st run - fail, 2nd, 3rd and 4th - pass
```

File: test/functional/quarantine-mode-tracker.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+module.exports = {
+    browsersPassingQuarantine: {},
+    browsersFailingQuarantine: {},
+
+    handleFailingSequence: function (ua) {
+        // NOTE: Failing sequence: 1st time - fail, 2nd - pass, 3rd and 4th - fail
```
Comment (VasilyStrelyaev) [inline test/functional/quarantine-mode-tracker.js]:
```
        // NOTE: Failing sequence: 1st run - fail, 2nd - pass, 3rd and 4th - fail
```

File: test/functional/quarantine-mode-tracker.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        var state = this.browsersFailingQuarantine[ua] || { step: 0 };
+
+        state.step++;
+
+        // NOTE: 4th step is the last one
```
Comment (VasilyStrelyaev) [inline test/functional/quarantine-mode-tracker.js]:
```
        // NOTE: 4th run is the last one
```

File: test/functional/quarantine-mode-tracker.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        this.browsersFailingQuarantine[ua] = state;
+
+        if (state.step === 2)
+            return 'pass';
+
+        return 'fail';
+    },
+
+    handlePassingSequence: function (ua) {
+        // NOTE: Passing sequence: 1st time - fail, 2nd, 3rd and 4th - pass
```
Comment (VasilyStrelyaev) [inline test/functional/quarantine-mode-tracker.js]:
```
        // NOTE: Passing sequence: 1st run - fail, 2nd, 3rd and 4th - pass
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by AndreyBelym** [comment]
Breaking change in `testcafe-hammerhead@6.0.1`: https://github.com/DevExpress/testcafe-hammerhead/commit/68efea1ed3718a3e9a7adb3bc20f63b88cdc9cc6

**Comment by AndreyBelym** [comment]
@testcafe-build-bot \retest

**Pull request merged**